pkg/auth/dependency/sso/sso.go

```diff
@@ -97,10 +97,14 @@ type AuthInfo struct {
 	ProviderUserID          string
 	ProviderUserProfile     map[string]interface{}
 	ProviderAccessTokenResp interface{}
-	ProviderAuthData        map[string]string
+	ProviderAuthData        ProviderAuthKeys
 	State                   State
 }
 
+type ProviderAuthKeys struct {
```

Comment:
IMO, `ProviderUserInfo` is a better name.

Reply:
This struct is intended to store only auth info (e.g. email) for matching with login IDs according to current SSO code. I agree changing it to `ProviderUserInfo` would be better, but then we should refactor the `DecodeXXX` functions in `AuthInfoProcessor` interface into single `DecodeUserInfo` function. We can leave this for later?

Reply:
It is good refactoring! When should we do this refactoring if not now?

Reply:
Will refactor SSO related code in #957